src/updaters/expo/app-json.ts

```diff
@@ -0,0 +1,91 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {jsonStringify} from '../../util/json-stringify';
+import {logger as defaultLogger, Logger} from '../../util/logger';
+import {DefaultUpdater, UpdateOptions} from '../default';
+import {Version} from '../../version';
+
+export interface AppJson {
+  expo: {
+    version: string;
+    ios: {
+      buildNumber: string;
+    };
+    android: {
+      versionCode: string;
+    };
+  };
+}
+
+export interface AppJsonOptions extends UpdateOptions {
+  expoSDKVersion: string;
+}
+
+/**
+ * This updates a React Natve Expo project app.json file's main, ios and android
+ * versions. All values except the `android.versionCode` are standard semver
+ * version numbers. For the `android.versionCode`, the semver number is used as
+ * the basis for the `versionCode`.
+ */
+export class AppJson extends DefaultUpdater {
+  expoSDKVersion: string;
+  constructor(options: AppJsonOptions) {
+    super(options);
+    this.expoSDKVersion = options.expoSDKVersion;
+  }
+  /**
+   * Given initial file contents, return updated contents.
+   */
+  updateContent(content: string, logger: Logger = defaultLogger): string {
+    const parsed = JSON.parse(content) as AppJson;
+
+    logger.info(
+      `updating Expo version from ${parsed.expo.version} to ${this.version}`
+    );
+    parsed.expo.version = this.version.toString();
+
+    logger.info(
+      `updating iOS version from ${parsed.expo.ios.buildNumber} to ${this.version}`
+    );
+    parsed.expo.ios.buildNumber = this.version.toString();
+
+    // Android versionCode
+    // https://developer.android.com/studio/publish/versioning#appversioning
+    let expoMajorVersion = 0;
+    try {
+      const expoVersion = Version.parse(this.expoSDKVersion);
+      expoMajorVersion = expoVersion.major;
+    } catch (e) {
+      // Rethrow with a nice error message.
+      throw new Error(
+        'Unable to determine the Expo SDK version for this project. Make sure that the expo package is installed for your project.'
+      );
+    }
+
+    // Implements the `versionCode` strategy described by Maxi Rosson
+    // @see https://medium.com/@maxirosson/versioning-android-apps-d6ec171cfd82
+    const versionCode =
+      expoMajorVersion * 10000000 +
+      this.version.major * 10000 +
+      this.version.minor * 100 +
+      this.version.patch;
+    logger.info(
+      `updating Android version from ${parsed.expo.android.versionCode} to ${versionCode}`
+    );
+    parsed.expo.android.versionCode = versionCode.toString();
```

Comment:
@chingor13 Here's the Android `versionCode` update strategy for Expo based projects that we discussed a couple weeks ago. As you can see, the `versionCode` is based on semver but also includes the Expo SDK version number. This is the implementation used on most Expo projects.